src/main/java/org/jenkinsci/plugins/workflow/steps/ErrorInfoStep.java

```diff
@@ -0,0 +1,193 @@
+ */
+public class ErrorInfoStep extends AbstractStepImpl {
+
+    public final Throwable error;
+
+    @DataBoundConstructor public ErrorInfoStep(Throwable error) {
+        this.error = error;
+    }
+
+    public static class Execution extends AbstractSynchronousStepExecution<ErrorInfo> {
+
+        private static final long serialVersionUID = 1;
+        @Inject private transient ErrorInfoStep step;
+        @StepContextParameter private transient FlowExecution execution;
+
+        @Override protected ErrorInfo run() throws Exception {
+            return new ErrorInfo(step.error, execution);
+        }
+
+    }
+
+    @Extension public static class DescriptorImpl extends AbstractStepDescriptorImpl {
+
+        public DescriptorImpl() {
+            super(Execution.class);
+        }
+
+        @Override public String getFunctionName() {
+            return "errorInfo";
+        }
+
+        @Override public String getDisplayName() {
+            return "Calculate information about an error";
+        }
+
+        // TODO blank config.jelly
+
+    }
+
+    public static class ErrorInfo implements Serializable {
+
+        private static final long serialVersionUID = 1;
+        private final Throwable error;
+        private transient FlowExecution execution;
+        private final FlowExecutionOwner executionOwner;
+
+        ErrorInfo(Throwable error, FlowExecution execution) {
+            this.error = error;
+            this.execution = execution;
+            executionOwner = execution.getOwner();
+        }
+
+        private FlowExecution getExecution() throws IOException {
+            if (execution == null) {
+                execution = executionOwner.get();
+            }
+            return execution;
+        }
+
+        /**
+         * Finds a node which threw this exception or one of its causes.
+         * Note that {@link Throwable#equals} is just pointer equality,
+         * which we cannot use since we may be loading deserialized exceptions,
+         * so we compare by stack trace instead.
+         */
+        private @CheckForNull FlowNode getNode() throws IOException {
+            final Set<String> stackTraces = new HashSet<>();
+            for (Throwable t = error; t != null; t = t.getCause()) {
+                stackTraces.add(Functions.printThrowable(t));
+            }
+            Predicate<FlowNode> threwException = new Predicate<FlowNode>() {
+                @Override
+                public boolean apply(FlowNode input) {
+                    if (input instanceof BlockEndNode) {
+                        return false;
+                    }
+                    ErrorAction a = input.getAction(ErrorAction.class);
+                    return (a != null && stackTraces.contains(Functions.printThrowable(a.getError())));
+                }
+            };
+            return new FlowScanner.DepthFirstScanner().findFirstMatch(getExecution(), threwException);
```

Comment:
`DepthFirstScanner` is a top-level type in the current upstream PR.

For a PR downstream of an evolving API PR, it is best to use timestamped snapshots in the Maven dependency, rather than `2.1-SNAPSHOT` which is vague.